integration-tests/suites/http_endpoint_availability.go

```diff
@@ -0,0 +1,47 @@
+package suites
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/stackrox/collector/integration-tests/pkg/collector"
+)
+
+type HttpEndpointAvailabilityTestSuite struct {
+	IntegrationTestSuiteBase
+	Port             int
+	CollectorOptions collector.StartupOptions
+	Endpoints        []string
+}
+
+func (s *HttpEndpointAvailabilityTestSuite) SetupSuite() {
+	s.StartCollector(false, &s.CollectorOptions)
+}
+
+func (s *HttpEndpointAvailabilityTestSuite) TestAvailability() {
+	for _, endpoint := range s.Endpoints {
+		url := fmt.Sprintf("http://localhost:%d%s", s.Port, endpoint)
+
+		fmt.Println(url)
+
+		resp, err := http.Get(url)
+		s.Require().NoError(err)
+		s.Require().NotNil(resp)
+		if resp != nil {
```

Comment:
Checking for nil twice seems a bit overkill.

Reply:
I think that the testing framework tries to continue execution as far as possible and reports a failure only after returning.

Reply:
I'm sure it doesn't, if you see that execution continues it is most likely due to the teardown section gets called even if an error occurred, but execution is stopped immediately.

Reply:
My bad, Require() stops immediately and Assert() does not.